### PR TITLE
Perfect sound for release

### PR DESCRIPTION
**Pull request by Stiffly**
The background music should now automatically change depending on what "state" the game is in.

### CODE REVIEW

File: src/Engine/Sound/SoundManager.cpp [hunk 1]
```diff
@@ -403,6 +407,39 @@ bool SoundManager::OnChangeBGM(const Events::ChangeBGM &e)
...
+
+bool SoundManager::OnPlayerDisconnected(const Events::PlayerDisconnected& e)
+{
+    // A player returned to the main menu. Easiest solution when we don't have states in the game. Not "PC".
+    if (m_CurrentBGMCombo != nullptr) {
+        if (getSourceState(m_CurrentBGMCombo->ALsource) == AL_PLAYING) {
+            stopSound(m_CurrentBGMCombo);
+        }
+    }
+    stopSound(m_CurrentBGMCombo);
```
Comment (wima12) [inline src/Engine/Sound/SoundManager.cpp]: I'm going to guess that this line should be removed?

File: src/Engine/Sound/SoundManager.cpp [hunk 2]
```diff
@@ -403,6 +407,39 @@ bool SoundManager::OnChangeBGM(const Events::ChangeBGM &e)
...
+    Events::ChangeBGM changeBGM;
+    changeBGM.FilePath = "Audio/BGM/MenuMusic.wav";
+    m_EventBroker->Publish(changeBGM);
+    return true;
+}
+
+
```
Comment (wima12) [inline src/Engine/Sound/SoundManager.cpp]: Listen for `Events::SetCamera` and 
`if (entity.Name() == "Overview_Camera_Start_Menu)"` set Menu music, 
`if (entity.Name() == "PickTeamCamera)"` set game background music.

Comment (Stiffly) [inline src/Engine/Sound/SoundManager.cpp]: Will do. Thanks

**Comment by Stiffly** [comment]
The menu music will be played when switching to menu camera. The match ambient sound will be played when pick team camera becomes active and continue to play throughout the match. :shipit:

**Pull request merged**